src/lightning_fabric/fabric.py

```diff
@@ -608,9 +609,17 @@ def log_dict(self, metrics: Dict[str, Any], step: Optional[int] = None) -> None:
 
         Args:
             metrics: A dictionary where the key is the name of the metric and the value the scalar to be logged.
+                Any :class:`torch.Tensor`s in the dictionary get detached from the graph automatically.
             step: Optional step number. Most Logger implementations auto-increment this value by one with every
                 log call. You can specify your own value here.
         """
+
+        def to_item(value: Tensor) -> int | float | bool:
```

Comment:
The new style annotations won't work without the __future__ import. But it doesn't work with libraries that inspect signatures (e.g. jsonarpgarse). Use good old Union

```suggestion
        def to_item(value: Tensor) -> Union[int, float, bool]:
```

Reply:
I copied this code from the utilities, it is like this there too 😅

Reply:
But you didn't copy the `from __future__ import annotations`. The problem is that it breaks https://github.com/Lightning-AI/lightning/issues/11205. So better not use it.

In `metrics_to_scalars` it's not a problem because it's not parsed by the CLI